test/model.js

```diff
@@ -580,6 +618,25 @@
     equal(newModel.destroy(), false);
   });
 
+  test("destroy will pass extra options to success callback", 1, function () {
+    var SpecialSyncModel = Backbone.Model.extend({
+      sync: function (method, model, options) {
+        _.extend(options, { specialSync: true });
+        return Backbone.Model.prototype.sync.call(this, method, model, options);
+      },
+      urlRoot: '/test'
+    });
+
+    var model = new SpecialSyncModel({ id: 'id' });
+
+    model.destroy({ success: onSuccess });
+    this.ajaxSettings.success();
+
+    function onSuccess(model, response, options) {
```

Comment:
> Not even in a test?

Nah, we want the source code and its tests to have a consistent style.

> This technique takes advantage of function hoisting to allow this to work

Right. Just move the `onSuccess` above the `destroy` call or inline it in the `options` object. Function hoisting is a bad part of JS and serves no real point here.